00X-tracing.md

```diff
@@ -0,0 +1,579 @@
+   * @returns True if the event was emitted; false if the event was not emitted because tracing
+   *   was not enabled for any of the event categories.
+   */
+  count(
+    category: string | string[],
+    name: string,
+    value: number | { [name: string]: number } | () => (number | { [name: string]: number })): boolean;
+}
+```
+
+## 3. Tracing listener APIs
+The tracing listener APIs enable JavaScript code to listen to near-real-time tracing events in
+specific categories, rather than waiting for the events be recorded to a file for offline analysis.
+Tracing events emitted in both JavaScript can C++ can be received by both JavaScript and C++
+listeners.
+
+Relatively few applications and libraries are expected to use the listener APIs (as compared
+to the much broader applicability of the emitter APIs). However, the ability to listen to tracing
+events could be very valuable to certain application-monitoring, profiling, and diagnostic tools
+and SDKs.
+
+To support listeners, the `Tracing` class (from the previous section) extends a new
+`CategoryEventEmitter` base class. That class is similar to `EventEmitter`, but does not extend
+`EventEmitter` because it has different behavior related to the multi-category support.
+`CategoryEventEmitter` is designed to be non-tracing-specific, so it could be useful in other
+situations where multi-category event support is desired, however no specific other use cases
+are known at this time.
+
+Listening works whether or not the category is being recorded; in other words, adding a listener
+automatically enables the category for listening mode (but not necessarily recording), and removing
+the last listener disables the category for listening. So when nobody is listening (or recording)
+no events are emitted and the cost of tracing emit calls is near zero.
+
+### 3.1 Example use
+```javascript
+// Single-category listening and emitting:
+tracing.on('example', e => { // e is a TracingEvent object
+  if (e.name === 'info') {
+    console.log('Got a tracing event: ' + e.args.message);
+  }
+});
+tracing.emit('example', 'info', 'Hello tracing!');
+
+// Multi-category listening and emitting:
+// The listener receives events emitted for *any* matching category.
+tracing.on(['example1', 'example2'] , e => { // e is a TracingEvent object
+  if (e.name === 'info') {
+    console.log('Got a tracing event: ' + e.args.message);
+  }
+});
+tracing.emit(['example2', 'example3'], 'info', 'Hello tracing!');
+```
+
+### 3.2 Listener API specification
+```typescript
+interface Tracing extends CategoryEventEmitter ...
+
+/**
+ * Enables emitting and listening to events in multiple categories.
+ *
+ * This interface has methods that are similar to EventEmitter, with an important difference:
+ * events may be emitted under multiple categories simultaneously instead of a single event name,
+ * and listeners may listen to multiple categories while still only being called once per event.
+ */
```

Comment:
If it is similar but different, it should not be called `EventEmitter`, but something else.